app/assets/src/components/visualizations/heatmap/Heatmap.js

```diff
@@ -253,6 +263,7 @@ export default class Heatmap {
       .append("g")
       .attr("class", cx(cs.dendogram, "columnDendogram"));
     this.gColumnMetadata = this.g.append("g").attr("class", cs.columnMetadata);
+    this.gCaption = this.g.append("g").attr("class", cs.captionContainer);
```

Comment:
This does not seem like the right place for this.

If it is for svg only we should add just before export and remove it afterwords. 
We could do this in `download` and `downloadAsPng`.
Regarding size below,  we could just extend the svg by the need totalCaptionHeight again before saving.
Do you foresee any problems with this approach?

Reply:
This is a great point. I've made the code change and it seems to be working. I now have functions `showPrintCaption` and `hidePrintCaption`.